### DISCUSSION TITLE
Parameter using variable

### DISCUSSION THREAD

**Discussion by torodecamargue** [nextfoam/baram#112] [General]
Hello,

Is it possible to add a source term using the temperature variable ? 

For example volumic power, P = coef x temp

 If yes, could you details the typo? 

Thank you

Julien

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu from NEXTFOAM.

As you told if the source term is added like P = coef x temp, then there is no big diefference between current setup.
I think you want to configure the temperature as a time dependent variable.
In that case, the users could configure a source term as Piecewise Linear or Polynomial.

Or do you want a source term which temperature changes irregularly?
In this case, the temperature of cell zone would be changed continuously and the source term would be changed continuously and irregulary.

I want to know your suggestion more.

Best regards

Donggyu kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

I want to use a volume source term depending of the temperature in a transient study. The change of the temperature should be smooth.

Do you think BaramFlow can handle this in the source term ? What should I write in the source term for the tempertaure? h? 
And mathematical sign could be used also? Please give me and example fo typo for P = coef x temperature [W]

Than you

Best regards

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu.
Well... it sounds like very complicated. I should discuss with our developers.
I'll consider it to implement.

Thank you

Best regards
